packages/react/src/toolbar/button/ToolbarButton.tsx

```diff
@@ -0,0 +1,126 @@
+'use client';
+import * as React from 'react';
+import PropTypes from 'prop-types';
+import { useComponentRenderer } from '../../utils/useComponentRenderer';
+import { BaseUIComponentProps } from '../../utils/types';
+import { useButton } from '../../use-button';
+import { CompositeItem } from '../../composite/item/CompositeItem';
+import type { ToolbarRoot, ToolbarItemMetadata } from '../root/ToolbarRoot';
+import { useToolbarRootContext } from '../root/ToolbarRootContext';
+import { useToolbarGroupContext } from '../group/ToolbarGroupContext';
+
+/**
+ * A button that can be used as-is or as a trigger for other components.
+ * Renders a `<button>` element.
+ *
+ * Documentation: [Base UI Toolbar](https://base-ui.com/react/components/toolbar)
+ */
+const ToolbarButton = React.forwardRef(function ToolbarButton(
+  props: ToolbarButton.Props,
+  forwardedRef: React.ForwardedRef<HTMLButtonElement>,
+) {
+  const {
+    className,
+    disabled: disabledProp = false,
+    focusableWhenDisabled = true,
+    render,
+    ...otherProps
+  } = props;
+
+  const { disabled: toolbarDisabled, orientation } = useToolbarRootContext();
+
+  const groupContext = useToolbarGroupContext(true);
+
+  const itemMetadata = React.useMemo(() => ({ focusableWhenDisabled }), [focusableWhenDisabled]);
+
+  const disabled = toolbarDisabled || (groupContext?.disabled ?? false) || disabledProp;
+
+  const { getButtonProps } = useButton({
+    buttonRef: forwardedRef,
+    disabled,
+    focusableWhenDisabled,
+    type: 'button',
+  });
+
+  const state: ToolbarButton.State = React.useMemo(
+    () => ({
+      disabled,
+      orientation,
+      focusable: focusableWhenDisabled,
+    }),
+    [disabled, focusableWhenDisabled, orientation],
+  );
+
+  const extraProps = React.useMemo(
```

Comment:
This doesn't have to be memoized as `useComponentRenderer` destructures it anyway and doesn't care about object reference.